From 5c318611978a7f9add5b889ad70e4af5b10a9c00 Mon Sep 17 00:00:00 2001
From: Edward Swarthout <Ed.Swarthout@nxp.com>
Date: Sat, 8 Feb 2020 12:53:07 -0600
Subject: [PATCH] QNNPACK: q8gemm/8x8-dq-aarch64-neon.S fix mov operand for gcc

Unlike clang, GNU assembler does not support 4s on neon mov, so use 16b.

Fixes:

8x8-dq-aarch64-neon.S: Assembler messages:
8x8-dq-aarch64-neon.S:657:
Error: operand mismatch -- `mov V8.4s,V9.4s'
Info:    did you mean this?
Info:            mov v8.8b, v9.8b
Info:    other valid variant(s):
Info:            mov v8.16b, v9.16b

Signed-off-by: Edward Swarthout <Ed.Swarthout@nxp.com>
---
 .../cpu/qnnpack/src/q8gemm/8x8-dq-aarch64-neon.S | 16 ++++++++--------
 1 file changed, 8 insertions(+), 8 deletions(-)

diff --git a/aten/src/ATen/native/quantized/cpu/qnnpack/src/q8gemm/8x8-dq-aarch64-neon.S b/aten/src/ATen/native/quantized/cpu/qnnpack/src/q8gemm/8x8-dq-aarch64-neon.S
index 7dc861110186a..60ad8d1d4b340 100644
--- a/aten/src/ATen/native/quantized/cpu/qnnpack/src/q8gemm/8x8-dq-aarch64-neon.S
+++ b/aten/src/ATen/native/quantized/cpu/qnnpack/src/q8gemm/8x8-dq-aarch64-neon.S
@@ -659,14 +659,14 @@ BEGIN_FUNCTION pytorch_q8gemm_dq_ukernel_8x8__aarch64_neon
 
     SUB x1, x1, 4
 
-    MOV V8.4s, V9.4s
-    MOV v10.4s, v11.4s
-    MOV v12.4s, V13.4s
-    MOV V14.4s, V15.4s
-    MOV V16.4s, V17.4s
-    MOV V18.4s, V19.4s
-    MOV V20.4s, V21.4s
-    MOV V22.4s, V23.4s
+    MOV V8.16b, V9.16b
+    MOV v10.16b, v11.16b
+    MOV v12.16b, V13.16b
+    MOV V14.16b, V15.16b
+    MOV V16.16b, V17.16b
+    MOV V18.16b, V19.16b
+    MOV V20.16b, V21.16b
+    MOV V22.16b, V23.16b
 
 5:
     CMP x1, 2
